File: android/app/src/main/java/com/magicforms/mobile/data/api/DeviceApi.kt
package com.magicforms.mobile.data.api

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json

class DeviceApi {
    private val json = Json { ignoreUnknownKeys = true }
    private val authApi = AuthApi()

    fun registerPushToken(bearerToken: String, pushToken: String, deviceId: String) {
        val body = RegisterBody(platform = "android", token = pushToken, deviceId = deviceId)
        authApi.post(ApiConfig.url("/api/v1/devices/register/"), json.encodeToString(body), bearerToken)
    }

    fun unregisterPushToken(bearerToken: String, pushToken: String?) {
        val body = UnregisterBody(token = pushToken)
        authApi.post(ApiConfig.url("/api/v1/devices/unregister/"), json.encodeToString(body), bearerToken)
    }

    @Serializable
    private data class RegisterBody(
        val platform: String,
        val token: String,
        @SerialName("device_id") val deviceId: String = "",
    )

    @Serializable
    private data class UnregisterBody(val token: String? = null)
}

File: ios/MagicFormsMobile/Services/DeviceAPI.swift
import Foundation
import UIKit

enum DeviceAPI {
    private struct RegisterBody: Encodable {
        let platform: String
        let token: String
        let deviceId: String

        enum CodingKeys: String, CodingKey {
            case platform
            case token
            case deviceId = "device_id"
        }
    }

    private struct UnregisterBody: Encodable {
        let token: String?
    }

    @MainActor
    static func registerPushToken(bearerToken: String, pushToken: String) async {
        let url = APIConfig.url(path: "/api/v1/devices/register/")
        let body = RegisterBody(
            platform: "ios",
            token: pushToken,
            deviceId: UIDevice.current.identifierForVendor?.uuidString ?? ""
        )
        guard let payload = try? JSONEncoder().encode(body) else { return }
        _ = try? await AuthAPI.postData(url: url, body: payload, token: bearerToken)
    }

    @MainActor
    static func unregisterPushToken(bearerToken: String, pushToken: String?) async {
        let url = APIConfig.url(path: "/api/v1/devices/unregister/")
        let body = UnregisterBody(token: pushToken)
        guard let payload = try? JSONEncoder().encode(body) else { return }
        _ = try? await AuthAPI.postData(url: url, body: payload, token: bearerToken)
    }
}
